### PR TITLE
Adding text about creating and verifying Teep messages

### PR DESCRIPTION
**Pull request by hannestschofenig**

### CODE REVIEW

File: draft-ietf-teep-protocol.xml [hunk 1]
```diff
@@ -227,32 +227,91 @@ otherwise.
 
     <section anchor="detailmsg" title="Detailed Messages Specification">
 
-    <t>The CBOR-encoded messages are protected by COSE, as described in CDDL format <xref target="I-D.ietf-cbor-cddl"/> below. 
+    <t>TEEP messages are protected by the COSE_Sign1 structure.
+      The TEEP protocol messages are described in CDDL format <xref target="I-D.ietf-cbor-cddl"/> below. 
     
  <figure>
  <artwork><![CDATA[
```
Comment (cabo) [inline draft-ietf-teep-protocol.xml:234]: This edit looks incomplete.  Please check the CDDL after changing it...

File: draft-ietf-teep-protocol.xml [hunk 2]
```diff
@@ -726,8 +785,28 @@ Value | Ciphersuite
       <xref target="ciphersuite"/>. 
       </t>
       </section> 
+      
+      <section title="CBOR Tag Registry">
+
+    <t>IANA is requested to register a CWT CBOR tag in the "CBOR Tags" registry
+      [IANA.CBOR.Tags] for use with Teep messages.</t>
```
Comment (cabo) [inline draft-ietf-teep-protocol.xml:792]: That is not the right registry..

File: draft-ietf-teep-protocol.xml [hunk 3]
```diff
@@ -227,32 +227,91 @@ otherwise.
...
+       supported or that are specified as being ignored when not
+       understood.
+   </t> 
+   
+   <t> Follow the steps specified in Section 4 of [RFC8152] ("Signing Objects") for
+          validating a COSE_Sign1 object. The COSE_Sign1 payload is the content 
+          of the Teep message.
+   </t>
+   
+   <t> Verify that the Teep message is a valid CBOR map and verify the fields of the
```
Comment (cabo) [inline draft-ietf-teep-protocol.xml:304]: Teep vs. TEEP, as per VI

File: draft-ietf-teep-protocol.xml [hunk 4]
```diff
@@ -227,32 +227,91 @@ otherwise.
...
-
-Msg_AuthEnc_Wrapper = [ * (COSE_Mac_Tagged / 
-                           COSE_Sign_Tagged / 
-                           COSE_Mac0_Tagged / 
-                           COSE_Sign1_Tagged)]
                        ]]></artwork>
           </figure>
      </t> 
 
+ <section title="Creating and Validating Teep Messages">
```
Comment (dthaler) [inline draft-ietf-teep-protocol.xml:246]:
```suggestion
 <section title="Creating and Validating TEEP Messages">
```
please make this change throughout

File: draft-ietf-teep-protocol.xml [hunk 5]
```diff
@@ -227,32 +227,91 @@ otherwise.
...
+<list style="numbers"> 
+
+   <t> Create a Teep message according to the description below and populate 
+       it with the respective content.</t>
+
+   <t>  Create a COSE Header containing the desired set of Header
+       Parameters.  The COSE Header MUST be valid per the [RFC8152]
+       specification.</t>
+
+   <t>  If the Teep message is signed create a COSE_Sign1 object
```
Comment (dthaler) [inline draft-ietf-teep-protocol.xml:262]: I think all TEEP messages MUST be signed, so there should be no "if" here

File: draft-ietf-teep-protocol.xml [hunk 6]
```diff
@@ -227,32 +227,91 @@ otherwise.
...
+<t>        
+<list style="numbers"> 
+
+   <t> Create a Teep message according to the description below and populate 
+       it with the respective content.</t>
+
+   <t>  Create a COSE Header containing the desired set of Header
+       Parameters.  The COSE Header MUST be valid per the [RFC8152]
+       specification.</t>
+
```
Comment (dthaler) [inline draft-ietf-teep-protocol.xml:261]: Need to explain here where encryption comes in... messages should be encrypted first when confidentiality is needed. and THEN signed.  (OTrP already worked that way.)

File: draft-ietf-teep-protocol.xml [hunk 7]
```diff
@@ -227,32 +227,91 @@ otherwise.
...
+          COSE_Sign1 object MUST be followed.</t>
+          
+   <t> Prepend the COSE object with the
+       Teep CBOR tag to indicate that the CBOR-encoded message is indeed a 
+       Teep message. </t> 
+       
+</list>
+</t>
+
+<t>Note: Teep messages are identified
```
Comment (dthaler) [inline draft-ietf-teep-protocol.xml:274]: What was this supposed to say?

File: draft-ietf-teep-protocol.xml [hunk 8]
```diff
@@ -227,32 +227,91 @@ otherwise.
...
+       and values whose syntax and semantics are both understood and
+       supported or that are specified as being ignored when not
+       understood.
+   </t> 
+   
+   <t> Follow the steps specified in Section 4 of [RFC8152] ("Signing Objects") for
+          validating a COSE_Sign1 object. The COSE_Sign1 payload is the content 
+          of the Teep message.
+   </t>
+
```
Comment (dthaler) [inline draft-ietf-teep-protocol.xml:303]: Need to deal with decryption here, if the message is encrypted

File: draft-ietf-teep-protocol.xml (hunk 1 above)
Comment (hannestschofenig) [reply draft-ietf-teep-protocol.xml:234]: More CDDL edits will follow from Akira

File: draft-ietf-teep-protocol.xml (hunk 6 above)
Comment (hannestschofenig) [reply draft-ietf-teep-protocol.xml:261]: This needs to be described in a dedicated paragraph

**Pull request merged**